src/mlpack/methods/preprocess/preprocess_imputer_main.cpp

```diff
@@ -150,8 +124,33 @@ static void mlpackMain()
       Log::Info << "Performing '" << strategy << "' imputation strategy "
           << "to replace '" << missingValue << "' on dimension " << dimension
           << "." << endl;
-
-      imputer.Impute(input, missingValue, dimension);
+      if (strategy == "mean")
+      {
+        Imputer<double, MapperType, MeanImputation<double>> imputer(info);
+        imputer.Impute(input, missingValue, dimension);
+      }
+      else if (strategy == "median")
+      {
+        Imputer<double, MapperType, MedianImputation<double>> imputer(info);
+        imputer.Impute(input, missingValue, dimension);
+      }
+      else if (strategy == "listwise_deletion")
+      {
+        Imputer<double, MapperType, ListwiseDeletion<double>> imputer(info);
+        imputer.Impute(input, missingValue, dimension);
+      }
+      else if (strategy == "custom")
+      {
+        CustomImputation<double> strat(customValue);
+        Imputer<double, MapperType, CustomImputation<double>> imputer(
+            info, strat);
+        imputer.Impute(input, missingValue, dimension);
+      }
+      else
+      {
+        Log::Fatal << "'" <<  strategy << "' imputation strategy does not exist"
+          << endl;
```

Comment:
This is a really minor style comment, but new lines should be indented two tabs (four spaces) from the previous line.  If you prefer I can handle this (and comments like it) after the merge; just let me know if you'd like me to do that.